packages/loot-design/src/components/modals/SelectLinkedAccounts.js

```diff
@@ -1,254 +1,200 @@
+                      chosenAccounts[item.account_id] === addAccountOption.value
+                        ? {
+                            id: addAccountOption.value,
+                            name: addAccountOption.label,
+                          }
+                        : localAccounts.find(
+                            acc => chosenAccounts[item.account_id] === acc.id,
+                          )
+                    }
+                    unlinkedAccounts={unlinkedAccounts}
+                    onSetLinkedAccount={onSetLinkedAccount}
+                  />
                 </View>
               )}
-            </View>
+            />
           </View>
-
           <View
             style={{
               flexDirection: 'row',
               justifyContent: 'flex-end',
-              marginTop: 30,
+              marginTop: 10,
             }}
           >
-            <Button style={{ marginRight: 10 }} onClick={modalProps.onClose}>
-              Cancel
-            </Button>
-            <Button primary onClick={onNext} disabled={!chosenAccounts.length}>
+            <Button
+              primary
+              onClick={onNext}
+              disabled={!Object.keys(chosenAccounts).length}
+            >
               Link accounts
             </Button>
           </View>
-        </View>
+        </>
       )}
     </Modal>
   );
 }
+
+function TableRow({
+  externalAccount,
+  linkedAccount,
+  chosenAccount,
+  unlinkedAccounts,
+  onSetLinkedAccount,
+}) {
+  const [focusedField, setFocusedField] = useState(null);
+
+  const chosenAccountOption = chosenAccount && {
+    value: chosenAccount.id,
+    label: chosenAccount.name,
+  };
+
+  const availableAccountOptions = [
+    ...unlinkedAccounts.map(acct => ({
+      value: acct.id,
+      label: acct.name,
+    })),
+    chosenAccount?.id !== addAccountOption.value && chosenAccountOption,
+    addAccountOption,
+  ].filter(Boolean);
+
+  return (
+    <Row style={{ backgroundColor: 'white' }}>
+      <Field width={200}>{externalAccount.name}</Field>
+      <Field
+        width="flex"
+        truncate={focusedField !== 'account'}
+        onClick={() => !linkedAccount && setFocusedField('account')}
+      >
+        {focusedField === 'account' ? (
+          <Autocomplete
+            autoFocus
+            options={availableAccountOptions}
+            onSelect={value => {
+              onSetLinkedAccount(externalAccount, value);
+            }}
+            onBlur={() => setFocusedField(null)}
+            value={chosenAccountOption}
+          />
+        ) : (
+          linkedAccount?.name || chosenAccount?.name
+        )}
+      </Field>
+      <Field width="flex">
+        {!linkedAccount &&
+          (chosenAccount ? (
+            <Button
+              onClick={() => {
+                onSetLinkedAccount(externalAccount, null);
+              }}
+              style={{ float: 'right' }}
+            >
+              Remove bank-sync
```

Comment:
Unlink? or maybe Stop syncing? Disconnect?

Reply:
Also, a thought: instead of having explicit buttons to start/stop syncing, could you instead be allowed to pick accounts from the dropdown, clear the account field if you want to stop syncing, then click cancel/save at the bottom to commit or revert all the changes at once?

Reply:
Closing the modal will cancel any selections made. The actual account link is persisted only when you click "link accounts" at the bottom.

If we remove the buttons then I'm afraid it will be tricky for the user to know that inside the table columns there are dropdowns that are exposed upon click. We have the same logic in the transaction table, but IMO there it's sort of implied.. whereas here the functionality might not be es easy to read without an explicit call-to-action.